base/file.jl

```diff
@@ -341,37 +354,47 @@ Obtain the path of a temporary directory (possibly shared with other processes).
 tempdir()
 
 """
-    tempname()
+    tempname(; prefix="$temp_prefix")
+
+Generate a unique temporary file path. A `prefix` can be provided in addition, which
+otherwise defaults to "$temp_prefix".
 
-Generate a unique temporary file path.
+*Note: The `prefix` may be truncated based on the limitations of the OS.*
```

Comment:
The tempname() is to be deprecated in future. It may be ideal to just drop support of prefix from tempname for now. You still would need to keep the prefix in Windows version or rename prefix based version to something else as that will be used internally by mktemp or mktempdir.